### PR TITLE
Neptune Gremlin QPT Enhancement for valueMap Traversal Queries

### PR DESCRIPTION
**Pull request by Trianz-Akshay**
*Issue #, if available:*

*Description of changes:*
**Overview**
This PR adds full valueMap() support for Gremlin Query Pushdown (QPT) in the Athena–Neptune connector.
Row writers now correctly handle both:

- valueMap()-style results (where property values are Lists), and
- project().by()-style results (scalars or nested Maps).

This avoids ClassCastException and enables consistent behavior across different Gremlin result shapes.

**Problem**
Gremlin QPT can return property data in different shapes depending on the traversal:
_valueMap()_ → properties are returned as Map<String, List<Object>>
Example: name -> [John]
_project().by() / by(valueMap())_ → properties can be scalars or Map values
Example: all_properties -> Map

Previously, row writers assumed a single result shape when reading field values.
This could lead to:

- ClassCastException
- Incorrect field handling when the result format differed

**Changes**
Row writers updated to use the normalizer
All extractors in the following classes now retrieve field values using:
FieldValueNormalizer.toValueList(...)
The extractors then read from the resulting list, ensuring compatibility with both valueMap() and project()-style results.
Updated classes:

1. CustomSchemaRowWriter – Custom schema vertex/edge row writing
2. VertexRowWriter – Vertex row writing
3. EdgeRowWriter – Edge row writing

Find attached document for [supported query types and limitations](https://github.com/user-attachments/files/25545370/Neptune_Gremlin_QPT_valueMap_Support_Updated.docx) as well as [functional test results](https://github.com/user-attachments/files/25545393/NEPTUNE_FUNCTIONAL_TEST.xlsx).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ejeffrli — DISMISSED** [review]

File: athena-neptune/src/test/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/CustomSchemaRowWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+            reader.setPosition(0);
+            if (!reader.isSet()) {
+                return null;
+            }
+            return reader.readObject();
+        }
+    }
+
+    @Test
+    public void writeRowTemplate_varchar_valueMapList_writesFirstElement() throws Exception
```
Comment (fal-bharadwaj) [inline athena-neptune/src/test/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/CustomSchemaRowWriterTest.java:108]: Test coverage for CustomSchemaRowWriter is ~57.7%, can we add more test classes to account for the missing code paths? Let's have test cases for scalar input, single element list input, null/empty etc.

**Review by fal-bharadwaj — CHANGES_REQUESTED** [review]

File: athena-neptune/src/test/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/CustomSchemaRowWriterTest.java (hunk 1 above)
Comment (VenkatasivareddyTR) [reply athena-neptune/src/test/java/com/amazonaws/athena/connectors/neptune/propertygraph/rowwriters/CustomSchemaRowWriterTest.java:108]: Hi @fal-bharadwaj, thanks for the review. The Neptune connector currently has low baseline coverage, which is reflected in this class. We have covered all new and updated code and added tests for missing paths here (coverage is 84.44% now). Since there is a separate task dedicated to increasing Neptune's overall coverage, we will add more tests to the remaining classes as part of that task.

**Review by fal-bharadwaj — APPROVED** [review]

**Review by burhan94 — COMMENTED** [review]
## Code Review

### ✅ Looks Good

New `FieldValueNormalizer` utility handles both valueMap() and project().by() results cleanly. Good refactoring of row writers to use List interface. Comprehensive tests. Good null handling with early returns.

**Review by hareeshkumar-t — APPROVED** [review]
